internal/translayer/dbuser/dbuser.go

```diff
@@ -0,0 +1,57 @@
+package dbuser
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+type Service struct {
+	admin.DatabaseUsersApi
+}
+
+func NewService(ctx context.Context, provider atlas.Provider, secretRef *types.NamespacedName, log *zap.SugaredLogger) (*Service, error) {
+	client, err := translayer.NewVersionedClient(ctx, provider, secretRef, log)
+	if err != nil {
+		return nil, err
+	}
+	return &Service{DatabaseUsersApi: client.DatabaseUsersApi}, nil
+}
+
+func (dus *Service) Get(ctx context.Context, db, projectID, username string) (*User, error) {
+	atlasDBUser, _, err := dus.GetDatabaseUser(ctx, projectID, db, username).Execute()
+	if err != nil {
+		if admin.IsErrorCode(err, atlas.UsernameNotFound) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return toK8sDatabaseUser(atlasDBUser)
+}
+
+func (dus *Service) Delete(ctx context.Context, db, projectID, username string) (bool, error) {
+	_, _, err := dus.DeleteDatabaseUser(ctx, projectID, db, username).Execute()
+	if err != nil {
+		if admin.IsErrorCode(err, atlas.UsernameNotFound) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (dus *Service) Create(ctx context.Context, au *User) error {
+	_, _, err := dus.CreateDatabaseUser(ctx, au.ProjectID, toAtlas(au)).Execute()
```

Comment:
what if we need data after creation, filled by the SDK or API endpoint?

Reply:
For this example I just tailored the calls to their actual usage. The creation was not using the object after the fact, so this suffices here.

If we needed to use the returned object from the API call we would use the conversion `fromAtlas...` and return it back to the consumer.